moto/iam/access_control.py

```diff
@@ -459,6 +518,25 @@ def _check_element_matches(self, statement_element: Any, value: str) -> bool:
         else:  # string
             return self._match(self._statement[statement_element], value) is not None
 
+    def _check_principal(self, principal: str) -> bool:
+        expected_principals = self._statement.get("Principal")
+        if not expected_principals:
+            return True
+
+        aws_expected_principals = expected_principals.get("AWS")
+        return (
+            aws_expected_principals == "*"
+            or principal == aws_expected_principals
+            or principal in aws_expected_principals
+        )
+
+    def _check_conditions(self, conditions: Dict[str, Dict[str, str]]) -> bool:
+        expected_conditions = self._statement.get("Condition")
+        if not expected_conditions:
+            return True
+
+        return conditions == expected_conditions
```

Comment:
There could be multiple 'expected' conditions, right? The policy could say something like:
```
                "Condition": {
                    "StringEquals": {
                        "sts:ExternalId": external_id,
                        "awsAccountId": ...
                    },
                   "StringLike": {
                       ....
                   }
                },
```
So this logic should be a bit more advanced - it can't be a simple comparison.

Reply:
Redesign conditions evaluation process with extra entities